.github/workflows/deploy.yml

```
@@ -1401,20 +1423,23 @@ jobs:
       uses: actions/upload-artifact@v4
       if: always()
       with:
-        name: test-results
+        name: test-results-shard-${{ matrix.shard }}
         path: |
           test-results-*.xml
-          
+
     - name: Set up Python 3.8 for gcovr
+      if: matrix.shard == 1
       uses: actions/setup-python@v4
```

actionlint (1.7.12)

[error] the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)